src/Fantomas/Context.fs

```diff
@@ -567,21 +567,21 @@ let internal leadingExpressionIsMultiline leadingExpression continuationExpressi
     let hasWriteLineEventsAfterExpression = contextAfterLeading.WriterEvents |>Seq.skip eventCountBeforeExpression |> Seq.exists (function | WriteLine _ -> true | _ -> false)
     continuationExpression hasWriteLineEventsAfterExpression contextAfterLeading
 
-let private expressionExceedsPageWidth beforeShort afterShort beforeLong afterLong expr (ctx: Context) =
+let private expressionExceedsWidth width ignoreShort beforeShort afterShort beforeLong afterLong expr (ctx: Context) =
     // if the context is already inside a ShortExpression mode, we should try the shortExpression in this case.
     match ctx.WriterModel.Mode with
-    | ShortExpression _ ->
+    | ShortExpression _ when not ignoreShort ->
```

Comment:
What's the deal with this `ignoreShort`?

Reply:
Used in `Context.fs` (notice `true` as second parameter to `expressionExceedsWidth`):
```
let internal sepSpaceOrIndentAndNlnIfExpressionExceedsMaxBindingWidthIgnoreShort expr (ctx:Context) =
    expressionExceedsWidth
        ctx.Config.MaxBindingWidth true
        sepSpace sepNone // before and after for short expressions
        (indent +> sepNln) unindent // before and after for long expressions
        expr ctx
```

Which is used by `CodePrinter.fs` in `genMemberBinding`:
```
        match e with
        | TypedExpr(Typed, e, t) ->
            genAttributesAndXmlDoc
            +> leadingExpressionIsMultiline prefix (fun prefixIsLong ->
                ifElse
                    prefixIsLong
                    (indent +> sepNln +> sepColon +> genType astContext false t +> sepNln +> !- "=" +> sepNln +> genExpr astContext e +> unindent)
                    (sepColon +> genType astContext false t +> sepEq +> sepSpaceOrIndentAndNlnIfExpressionExceedsMaxBindingWidthIgnoreShort (genExpr astContext e)))
        | e ->
            genAttributesAndXmlDoc
            +> leadingExpressionIsMultiline prefix
                (fun prefixIsLong ->
                    ifElse
                        prefixIsLong
                        (indent +> sepNln +> !- "=" +> sepNln +> genExpr astContext e +> unindent)
                        (sepEq +> sepSpaceOrIndentAndNlnIfExpressionExceedsMaxBindingWidthIgnoreShort (genExpr astContext e)))
```

The `Context` seems to be in `ShortExpression` mode when generating member bindings which results in `MaxBindingWidth` being ignored because `expressionExceedsWidth` will match `ShortExpression`.